clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -152,7 +152,7 @@ public String socketDescription() {
 
     public void setSend(Send send) {
         if (this.send != null)
-            throw new IllegalStateException("Attempt to begin a send operation with prior send operation still in progress.");
+            throw new IllegalStateException("Attempt to begin a send operation with prior send operation still in progress, send destination is " + send.destination());
```

Comment:
Nit: maybe we should use the channel id and use the connection id terminology that we are using in other log messages. What do you think?